src/Common/Core/Impl/OS/Win32Process.cs

```diff
@@ -0,0 +1,130 @@
+
+            STARTUPINFO si = new STARTUPINFO();
+            si.cb = Marshal.SizeOf(typeof(STARTUPINFO));
+
+            /* 
+            When a process is started using CreateProcessAsUser function, the process will be started into a windowstation 
+            and desktop combination based on the value of lpDesktop in the STARTUPINFO structure parameter:
+            lpDesktop = "<windowsta>\<desktop>"; the system will try to start the process into that windowstation and desktop.
+            lpDesktop = NULL; the system will try to use the same windowstation and desktop as the calling process if the system is associated with the interactive windowstation.
+            lpDesktop = <somevalue>; the system will create a new windowstation and desktop that you cannot see.
+            lpDesktop = ""; it will either create a new windowstation and desktop that you cannot see, or if one has been created by means of a prior call by using the same access token, the existing windowstation and desktop will be used.
+            */
+            si.lpDesktop = "";
+
+            IntPtr stdinRead, stdinWrite, stdoutRead, stdoutWrite, stderrorRead, stderrorWrite;
+
+            SECURITY_ATTRIBUTES sa = default(SECURITY_ATTRIBUTES);
+            sa.nLength = Marshal.SizeOf(sa);
+            sa.lpSecurityDescriptor = IntPtr.Zero;
+            sa.bInheritHandle = true;
+
+            if (!CreatePipe(out stdinRead, out stdinWrite, ref sa, 0)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            if (!SetHandleInformation(stdinWrite, HANDLE_FLAGS.INHERIT, HANDLE_FLAGS.None)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            if (!CreatePipe(out stdoutRead, out stdoutWrite, ref sa, 0)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            if (!SetHandleInformation(stdoutRead, HANDLE_FLAGS.INHERIT, HANDLE_FLAGS.None)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            if (!CreatePipe(out stderrorRead, out stderrorWrite, ref sa, 0)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            if (!SetHandleInformation(stderrorRead, HANDLE_FLAGS.INHERIT, HANDLE_FLAGS.None)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            si.dwFlags = STARTF_USESTDHANDLES;
+            si.hStdInput = stdinRead;
+            si.hStdOutput = stdoutWrite;
+            si.hStdError = stderrorWrite;
+
+            SECURITY_ATTRIBUTES processAttr = CreateSecurityAttributes(winIdentity == null ? WellKnownSidType.AuthenticatedUserSid : WellKnownSidType.NetworkServiceSid);
+            SECURITY_ATTRIBUTES threadAttr = CreateSecurityAttributes(winIdentity == null ? WellKnownSidType.AuthenticatedUserSid : WellKnownSidType.NetworkServiceSid);
+
+            PROCESS_INFORMATION pi;
```

Comment:
MSDN says:

> Handles in PROCESS_INFORMATION must be closed with CloseHandle when they are no longer needed.

> If the function succeeds, be sure to call the CloseHandle function to close the hProcess and hThread handles when you are finished with them. Otherwise, when the child process exits, the system cannot clean up the process structures for the child process because the parent process still has open handles to the child process. 

Given the other comment regarding the use of `Process`, I think this class just needs to become non-static, and manage the lifetime of these handles, just like `Process` does.